    Graded Homework #2 covers the topics in Weeks 1, 2, 3, 4, 5 and 6 and is worth 10% of your overall grade. 
    You may work on the homework for as long as you like within the given window. You are allowed only ONE attempt. 
    Please note that as long as you don't click "submit" for a question, you can change your answer as many times as needed. 
    You should only click "submit" when you are completely finished with the question and ready to submit your answer for grading. 

    Good luck!




### Question 1
    5.0 points possible (graded, results hidden)

    In a linear regression problem, we are using “R-squared” to measure goodness-of-fit.
    We add a feature (variable) in linear regression model and retrain the same model. Which of the following option is true?

        1. If R Squared increases, this variable is significant.
        2. If R Squared decreases, this variable is not significant.
        3. Individually R squared cannot tell about variable importance. We can’t say anything about it right now.
        4. None of these. 

#### Solution: 3

    “R squared” individually can’t tell whether a variable is significant or not because each time when we add a feature,
    “R squared” can either increase or stay constant. But, it is not true in case of “Adjusted R squared” (increases when features found to be significant).

### Question 2
    5.0 points possible (graded, results hidden)

    A correlation between age and health of a person found to be -1.09. On the basis of this you would tell the doctors that:

        1. Age is a good predictor of health
        2. Health is a good predictor of age
        3. Age is a poor predictor of health
        4. None of these 

#### Solution : 4

    None of These
    
    Correlation value cannot exceed absolute value of 1.

### Question 3

    5.0 points possible (graded, results hidden)
    
    You work for a bank where you are trying to predict the probability of default of a customer based on FICO score and annual income.
    Which of the following problems can arise while using a multiple linear regression model?
    
        1. There exists homoskedasticity in the model.
        2. The model can produce predicted probabilities that are less than zero and greater than one.
        3. The model leads to the omitted variable bias as only two independent factors can be included in the model.
        4. The model leads to an overestimation of the effect of independent variables on the dependent variable. 

_______________________________________________________________________________________________

    We try to build a model for NBA players’ salary.
    
    Download the dataset nba2017.csv from here: https://www.dropbox.com/s/pe3urv1mv9s8mwb/nba2017%20%281%29.csv?dl=0

    Load the dataset using the code nba = read.csv("nba2017.csv", header = TRUE).
    
    Now we take a closer look at the data set. There are four variables salary, Ht(Height), Exp(Experience) and expsq(the square of Experience).
    
    First, build a model using salary as the response and Ht and Exp as variables and denote it as Model_1.
    
    Build a second model using log(salary) as the response and Ht and Exp as variables, we denote it as Model_2.


In [3]:
# Code
library(tidyverse) 

# Reading data
nba = read.csv("nba2017.csv", header = TRUE)

In [6]:
model_1 <- lm(formula = Salary ~ Ht + Exp , data = nba)
model_1
summary(model_1)


Call:
lm(formula = Salary ~ Ht + Exp, data = nba)

Coefficients:
(Intercept)           Ht          Exp  
  -12353214      2253985       677390  



Call:
lm(formula = Salary ~ Ht + Exp, data = nba)

Residuals:
      Min        1Q    Median        3Q       Max 
-11584274  -2907609  -1371039   1880620  19884672 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -12353214    5785552  -2.135   0.0332 *  
Ht            2253985     874758   2.577   0.0103 *  
Exp            677390      61100  11.087   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 5728000 on 496 degrees of freedom
Multiple R-squared:  0.2042,	Adjusted R-squared:  0.201 
F-statistic: 63.65 on 2 and 496 DF,  p-value: < 2.2e-16


In [7]:
model_2 <- lm(formula = log(Salary) ~ Ht + Exp , data = nba)
model_2
summary(model_2)


Call:
lm(formula = log(Salary) ~ Ht + Exp, data = nba)

Coefficients:
(Intercept)           Ht          Exp  
     9.3942       0.6889       0.1674  



Call:
lm(formula = log(Salary) ~ Ht + Exp, data = nba)

Residuals:
    Min      1Q  Median      3Q     Max 
-6.0100 -0.6698  0.2475  1.0030  2.6381 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  9.39419    1.39946   6.713 5.24e-11 ***
Ht           0.68885    0.21159   3.256  0.00121 ** 
Exp          0.16738    0.01478  11.325  < 2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.385 on 496 degrees of freedom
Multiple R-squared:  0.2151,	Adjusted R-squared:  0.2119 
F-statistic: 67.96 on 2 and 496 DF,  p-value: < 2.2e-16


### Question 4

    5.0 points possible (graded, results hidden)

    For Model_1, what is the interpretation for the coefficient of height?

    1. One unit increase in height increases salary by 2253985 units
    2. One unit increase in height increases salary by 874758 units
    3. One unit increase in height decreases salary by 677390 units
    4. One unit increase in height increases salary by 677390 units 

#### Solution : 1

### Question 5

    5.0 points possible (graded, results hidden)

    For Model_2, what is the interpretation for the coefficient of height?

        1. When height increases by 1%, salary increases by 68.89%
        2. When height increases by 1 unit, salary increases by 0.6889%
        3. When height increases by 1%, salary increases by 0.6889 units
        4. When height increases by 1 unit, salary increases by 68.89% 

#### Solution 4:
    Property of Log linear model. 
